.github/workflows/beta-release-on-label.yml

```diff
@@ -0,0 +1,136 @@
+name: Publish Beta Package
+
+# This workflow is triggered on a label being added to a PR, and will publish a beta version of the bundle package e.g. @guardian/commercial-bundle-v1.0.1-beta.0
+# It gets the last version from the tags, bumps the prerelease version and publishes to npm with the next tag
+# if the current version is 1.0.0, the next beta version will be 1.0.1-beta.0, then 1.0.1-beta.1, etc.
+# semanic-release is not used here as it's very opinionated and refuses to work with non-semver versions or pull requests
+on:
+  pull_request:
+    types: [labeled]
+
+jobs:
+  has-permission:
+    name: Check permissions
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Check if user has admin access"
+        uses: "lannonbr/repo-permission-check-action@2.0.2"
+        with:
+          permission: "admin"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  bundle-release:
+    name: "@guardian/commercial-bundle"
+    runs-on: ubuntu-latest
+    needs: has-permission
+    if: github.event.label.name == '[beta] @guardian/commercial-bundle'
+    defaults:
+      run:
+        working-directory: ./bundle
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Setup node
+        uses: guardian/actions-setup-node@main
+
+      - name: Install dependencies
+        uses: bahmutov/npm-install@v1
+
+      - name: Build package
+        run: yarn build
+
+      - name: Grab last version
+        run: yarn version --new-version $(git tag -l | grep @guardian/commercial-bundle-\* |  sed -E 's|.*([0-9]+\.[0-9]+\.[0-9]+(-beta.[0-9]+)?)|\1|g' | tail -1) --no-git-tag-version
```

Comment:
😎 I'm just wondering if `tail -1` will get the last (aka _numeric_) tag from 'git tag -l' or just the last version that was tagged if that makes sense. If this happens on PRs could there be collisions?

Reply:
Also if this script is getting a bit gnarly then it might be worth moving to a fully fledged node script?

Reply:
Not sure I follow the question 🙈?

Reply:
Oh I am wondering if there is a race condition where we do tag version + 1 so that versions can get bumped out of numerical order. But I'm fine with this as is.

Reply:
hmm, seems quite unlikely, only if 2 PRs were tagged within a couple minutes of each other, in that case can just rerun I think